<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ClothSimulation</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QWidget" name="simulationWidget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="input">
         <property name="title">
          <string>Simulation Inputs</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QGroupBox" name="cloth_input">
            <property name="title">
             <string>Cloth</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="5" column="0">
              <widget class="Line" name="line_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="Line" name="line_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <layout class="QGridLayout" name="dimensions_grid">
               <item row="1" column="0">
                <layout class="QGridLayout" name="dimensions_input">
                 <item row="0" column="0">
                  <layout class="QFormLayout" name="formLayout">
                   <item row="0" column="0">
                    <widget class="QLabel" name="cloth_width_label">
                     <property name="text">
                      <string>width</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="cloth_width">
                     <property name="minimum">
                      <double>0.100000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>30.000000000000000</double>
                     </property>
                     <property name="value">
                      <double>1.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="0" column="1">
                  <layout class="QFormLayout" name="formLayout_2">
                   <item row="0" column="0">
                    <widget class="QLabel" name="cloth_height_label">
                     <property name="text">
                      <string>height</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="cloth_height">
                     <property name="minimum">
                      <double>0.100000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>30.000000000000000</double>
                     </property>
                     <property name="value">
                      <double>1.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="dimensions_label">
                 <property name="text">
                  <string>Dimensions</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="4" column="0">
              <layout class="QGridLayout" name="rotate_cloth_grid">
               <item row="0" column="0">
                <widget class="QLabel" name="rotate_cloth_label">
                 <property name="text">
                  <string>Rotate</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <layout class="QGridLayout" name="rotate_cloth_input">
                 <item row="2" column="0">
                  <layout class="QFormLayout" name="formLayout_5">
                   <item row="0" column="0">
                    <widget class="QLabel" name="rotate_cloth_angle_label">
                     <property name="text">
                      <string>angle</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSpinBox" name="rotate_cloth_angle">
                     <property name="minimum">
                      <number>-360</number>
                     </property>
                     <property name="maximum">
                      <number>360</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="2" column="1">
                  <layout class="QFormLayout" name="formLayout_6">
                   <item row="0" column="0">
                    <widget class="QLabel" name="rotate_cloth_axis_label">
                     <property name="text">
                      <string>axis</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QComboBox" name="rotate_cloth_axis">
                     <property name="enabled">
                      <bool>true</bool>
                     </property>
                     <property name="editable">
                      <bool>false</bool>
                     </property>
                     <property name="minimumContentsLength">
                      <number>1</number>
                     </property>
                     <item>
                      <property name="text">
                       <string>X</string>
                      </property>
                     </item>
                     <item>
                      <property name="text">
                       <string>Y</string>
                      </property>
                     </item>
                     <item>
                      <property name="text">
                       <string>Z</string>
                      </property>
                     </item>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item row="0" column="1" rowspan="8">
              <widget class="QScrollBar" name="verticalScrollBar">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="Line" name="line">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <layout class="QGridLayout" name="num_particle_grid">
               <item row="2" column="0">
                <layout class="QGridLayout" name="num_particles_input">
                 <item row="4" column="0">
                  <layout class="QFormLayout" name="formLayout_3">
                   <item row="0" column="0">
                    <widget class="QLabel" name="num_particles_x_label">
                     <property name="text">
                      <string>x</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSpinBox" name="num_particles_x">
                     <property name="minimum">
                      <number>2</number>
                     </property>
                     <property name="maximum">
                      <number>30</number>
                     </property>
                     <property name="value">
                      <number>5</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="4" column="1">
                  <layout class="QFormLayout" name="formLayout_4">
                   <item row="0" column="0">
                    <widget class="QLabel" name="num_particles_y_label">
                     <property name="text">
                      <string>y</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSpinBox" name="num_particles_y">
                     <property name="minimum">
                      <number>2</number>
                     </property>
                     <property name="maximum">
                      <number>30</number>
                     </property>
                     <property name="value">
                      <number>5</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="num_particles_label">
                 <property name="frameShadow">
                  <enum>QFrame::Plain</enum>
                 </property>
                 <property name="lineWidth">
                  <number>1</number>
                 </property>
                 <property name="midLineWidth">
                  <number>0</number>
                 </property>
                 <property name="text">
                  <string>Number of particles</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::AutoText</enum>
                 </property>
                 <property name="indent">
                  <number>-1</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="7" column="0">
              <layout class="QGridLayout" name="default_button_layout">
               <item row="0" column="1">
                <widget class="QPushButton" name="set_cloth_default_button">
                 <property name="text">
                  <string>Set to default</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item row="6" column="0">
              <layout class="QGridLayout" name="translate_cloth_grid">
               <item row="1" column="0">
                <layout class="QGridLayout" name="position_grid">
                 <item row="2" column="0">
                  <layout class="QFormLayout" name="formLayout_7">
                   <item row="0" column="0">
                    <widget class="QLabel" name="translate_x_label">
                     <property name="text">
                      <string>x</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="translate_x">
                     <property name="minimum">
                      <double>-100.000000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>100.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="2" column="1">
                  <layout class="QFormLayout" name="formLayout_8">
                   <item row="0" column="0">
                    <widget class="QLabel" name="translate_y_label">
                     <property name="text">
                      <string>y</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="translate_y">
                     <property name="minimum">
                      <double>-100.000000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>100.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="2" column="2">
                  <layout class="QFormLayout" name="formLayout_9">
                   <item row="0" column="0">
                    <widget class="QLabel" name="translate_z_label">
                     <property name="text">
                      <string>z</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QDoubleSpinBox" name="translate_z">
                     <property name="minimum">
                      <double>-100.000000000000000</double>
                     </property>
                     <property name="maximum">
                      <double>100.000000000000000</double>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="position_label">
                 <property name="lineWidth">
                  <number>1</number>
                 </property>
                 <property name="text">
                  <string>Translate</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="simulate_button">
            <property name="text">
             <string>Start Simulation</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stop_simulation_button">
            <property name="text">
             <string>Pause Simulation</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="restart_simulation_button">
            <property name="text">
             <string>Restart Simulation</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
